packages/react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -377,7 +377,7 @@ const DetailsListInner: React.ComponentType<IDetailsListInnerProps> = (
               groupNestingDepth,
               indentWidth,
               selection,
-              selectionMode,
+              selectionMode: checkboxVisibility !== CheckboxVisibility.hidden ? selectionMode : SelectionMode.none,
```

Comment:
It's perfectly acceptable to allow selection but hide the checkboxes!

Reply:
That's what I thought too. But in the case the `onRenderHeader` is **not** overridden by the consumer the `selectionMode` is overriden based on `checkboxVisibility` - [[link](https://github.com/microsoft/fluentui/blob/244770a98640cae0813270522a242a7c4a23dcb7/packages/react/src/components/DetailsList/DetailsList.base.tsx#L570)]

![image](https://user-images.githubusercontent.com/37994247/118942538-71b38c00-b970-11eb-9065-d719dfdbe85c.png)

More details on the issue is capture here - #18099 (including codepen links)

Allowing selection when `checkboxVisibility` is set to hidden doesn't seem to be allowed today for default render. Are you proposing we fix that instead?